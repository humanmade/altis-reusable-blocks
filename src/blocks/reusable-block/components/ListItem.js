import PropTypes from 'prop-types';

import { Button } from '@wordpress/components';
import Icon from './Icon';

const ListItem = ( { id, onClick, onHover, title, ...props } ) => (
	<li className="block-editor-block-types-list__list-item">
		<Button
			className="block-editor-block-types-list__item"
			onClick={ ( e ) => {
				e.preventDefault();
				onClick();
			} }
			onFocus={ () => onHover( id ) }
			onMouseEnter={ () => onHover( id ) }
			onMouseLeave={ () => onHover( null ) }
			onBlur={ () => onHover( null ) }
			{ ...props }
		>
			<span className="block-editor-block-types-list__item-icon">
				<Icon />
			</span>
			<span className="block-editor-block-types-list__item-title">
				{ title }
			</span>
		</Button>
	</li>
);

ListItem.propTypes = {
	id: PropTypes.number.isRequired,
	onClick: PropTypes.func.isRequired,
	onHover: PropTypes.func.isRequired,
	title: PropTypes.string.isRequired,
};

export default ListItem;
