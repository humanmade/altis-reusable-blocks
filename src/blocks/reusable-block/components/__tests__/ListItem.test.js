import { shallow } from 'enzyme';

import ListItem from '../ListItem';

jest.mock( '../Icon', () => mockComponent( 'Icon' ) );

describe( '<ListItem />', () => {

	test( 'renders as expected', () => {
		const result = shallow(
			<ListItem id={ 42 } onClick={ () => {} } onHover={ () => {} } title="Title" />
		);

		expect( result ).toMatchSnapshot();
	} );

} );
