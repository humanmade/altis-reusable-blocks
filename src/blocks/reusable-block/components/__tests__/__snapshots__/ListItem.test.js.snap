// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders as expected 1`] = `
<li
  className="block-editor-block-types-list__list-item"
>
  <ForwardRef(Button)
    className="block-editor-block-types-list__item"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="block-editor-block-types-list__item-icon"
    >
      <Icon />
    </span>
    <span
      className="block-editor-block-types-list__item-title"
    >
      Title
    </span>
  </ForwardRef(Button)>
</li>
`;
